Code/Framework/AzToolsFramework/AzToolsFramework/UI/DocumentPropertyEditor/DocumentPropertyEditor.cpp

```diff
@@ -583,31 +631,106 @@ namespace AzToolsFramework
         }
     }
 
-    AZ::Dom::Value DocumentPropertyEditor::GetDomValueForRow(DPERowWidget* row) const
+    void DocumentPropertyEditor::SetSavedExpanderStateForRow(DPERowWidget* row, DocumentPropertyEditor::ExpanderState expanderState)
     {
-        AZStd::vector<size_t> reversePath;
-        const DPERowWidget* thisRow = row;
-        const DPERowWidget* parentRow = thisRow->m_parentRow;
+        // Get the index of each dom child going up the chain. We can then reverse this
+        // and use these indices to mark a path to expanded/collapsed nodes
+        AZStd::vector<size_t> reversePath = GetPathToRoot(row);
```

Comment:
I'm good with this as-is, but I think `DomPrefixTree` can definitely make your life easier if you ever need to do something like this, it handles the "look up the nearest ancestor that explicitly has a value" behavior here.

Reply:
noted, thanks